t/t1092-sparse-checkout-compatibility.sh

```diff
@@ -692,6 +739,13 @@ test_expect_success 'sparse-index is not expanded' '
 	echo >>sparse-index/untracked.txt &&
 	ensure_not_expanded add . &&
 
+	ensure_not_expanded reset --hard &&
+	echo >>sparse-index/README.md &&
+	ensure_not_expanded reset --mixed &&
+	ensure_not_expanded reset --keep update-deep &&
+	ensure_not_expanded reset --merge rename-out-to-out &&
+	ensure_not_expanded reset --hard &&
+
```

Comment:
These could also be sprinkled into the necessary commits that make their lines possible.

I do notice that your examples for `--mixed` and `--hard` imply that your `<treeish>` is `HEAD`, which might not be the most interesting case. It might be good to do a cycle of the branches that do interesting things.


```
for ref in update-deep update-folder1 update-folder2 update-deep
do
ensure_not_expanded reset --mixed $ref
ensure_not_expanded reset --hard $ref
done &&
```

(I didn't include tabs because of the web editor.)

Reply:
Added this test loop for `--hard` and `--mixed` (both added in their relevant commits).